src/read.rs

```diff
@@ -468,7 +469,7 @@ impl<'a, T: Deserialize> NpyData<'a, T> {
     /// layout of the data, so if you want to correctly read multi-dimensional arrays you should
     /// switch to [`NpyFile`].
     pub fn to_vec(&self) -> Vec<T> {
-        let mut reader = self.inner.reader().clone();
+        let mut reader = self.inner.reader().deref().clone();
```

Comment:
Hmm.  I assume this is to clean up the "Clone of a reference" warning that I recall seeing on the first PR?  I'm not sure I understand why adding `.deref()` gets rid of it though.  Why would this warning warn only on `&mut &[T] -> &[T]` but not on `&&[T] -> &[T]`???

I figured the "proper" solution would just be

```rust
let mut reader = *self.inner.reader();
```
though I put proper in air quotes because I still think the original makes it more clear.  `let mut x = some_mut_borrow.clone()` very clearly communicates, "I'm making a temporary copy of something mutable so I can mutate that copy independently".  `let mut x = *some_mut_borrow` misdirects the reader into focusing too much on the fact that `some_mut_borrow` is a borrow, rather than on us copying it.

I'm going to post on URLO about this...